index.d.ts

```diff
@@ -30,6 +30,11 @@ export interface Options {
 	 * The registry URL is by default inferred from the npm defaults and `.npmrc`. This is beneficial as `package-json` and any project using it will work just like npm. This option is **only** intended for internal tools. You should **not** use this option in reusable packages. Prefer just using `.npmrc` whenever possible.
 	 */
 	readonly registryUrl?: string;
+
+	/**
+	 * Overwrite the `agent` that is passed down to [`got`](https://github.com/sindresorhus/got#agent). This might be useful to add [proxy support](https://github.com/sindresorhus/got#proxies).
+	 */
+	readonly agent?: Object;
```

Comment:
This could be better typed. Maybe just borrow the type of that Got option? https://github.com/DefinitelyTyped/DefinitelyTyped/blob/b1a104ea4f8b2b4ef1460d5b66aa7021dd609ebe/types/got/index.d.ts#L178

Reply:
Unfortunately I have no experience in writing type definitions, but I tried to copy over what seems relevant to match the Got definition.

Reply:
(Hopefully no one ever looks at the diff of the failing commit)